kong/router.lua

```diff
@@ -512,7 +512,8 @@ end
 do
   local matchers = {
     [MATCH_RULES.HOST] = function(route_t, ctx)
-      local host = route_t.hosts[ctx.hits.host or ctx.req_host]
+      local host = route_t.hosts[ctx.req_host] or
+                   route_t.hosts[ctx.hits.host]
```

Comment:
Here, we are ignoring the matched hint, which is thus rendered useless. In which case we could simply remove it. This change can be detrimental in some scenarios.

Reply:
What do you mean? If you remove `route_t.hosts[ctx.hits.host]` the tests will not pass, and if you remove `route_t.hosts[ctx.req_host]` the tests won't pass either. So it is not ignored.